amstrax/plugins/radon_peaks.py

```diff
@@ -0,0 +1,454 @@
+
+        s0 = np.argmax(b >= np.int16(bl))
+        ref = slice(max(0, s0-reference_length), s0)
+
+        if (b[ref] * to_pe[ch] > 1).sum() < min_reference_length:
+            # the pulse is saturated, but there are not enough reference samples to get a good ratio
+            # This actually distinguished between S1 and S2 and will only correct S2 signals
+            continue
+        if (b_sumwf[ref] > 1).sum() < min_reference_length:
+            # the same condition applies to the waveform model
+            continue
+        if np.sum(b[ref]) * to_pe[ch] / np.sum(b_sumwf[ref]) > 1:
+            # The pulse is saturated, but insufficient information is available in the other channels
+            # to reliably reconstruct it
+            continue
+
+        scale = np.sum(b[ref]) / np.sum(b_sumwf[ref])
+
+        # Loop over the record indices of the saturated channel (saved in b_index buffer)
+        for record_i in b_index[ch]:
+            if record_i == -1:
+                break
+            r = records[record_i]
+            r_slice, b_slice = strax.overlap_indices(
+                r['time'] // dt, r['length'],
+                p['time'] // dt + s0,  p['length'] * p['dt'] // dt - s0)
+
+            if r_slice[1] == r_slice[0]:  # This record proceeds saturation
+                continue
+            b_slice = b_slice[0] + s0, b_slice[1] + s0
+
+            # First is finding the highest point in the desaturated record
+            # because we need to bit shift the whole record if it exceeds int16 range
+            apax = scale * max(b_sumwf[slice(*b_slice)])
+
+            if np.int32(apax) >= 2**15:  # int16(2**15) is -2**15
+                bshift = int(np.floor(np.log2(apax) - 14))
+
+                tmp = r['data'].astype(np.int32)
+                tmp[slice(*r_slice)] = b_sumwf[slice(*b_slice)] * scale
+
+                r['area'] = np.sum(tmp)  # Auto covert to int64
+                r['data'][:] = np.right_shift(tmp, bshift)
+                r['amplitude_bit_shift'] += bshift
+            else:
+                r['data'][slice(*r_slice)] = b_sumwf[slice(*b_slice)] * scale
+                r['area'] = np.sum(r['data'])
+
+
+@numba.njit(cache=True, nogil=True)
+def hit_max_sample(records, hits):
+    """Return the index of the maximum sample for hits"""
+    result = np.zeros(len(hits), dtype=np.int16)
+    for i, h in enumerate(hits):
+        r = records[h['record_i']]
+        w = r['data'][h['left']:h['right']]
+        result[i] = np.argmax(w)
+    return result
+
+
+@export
+@strax.takes_config(
+    # PMT pulse processing options
+    strax.Option(
+        'save_outside_hits',
+        default=(3, 20),
+        help='Save (left, right) samples besides hits; cut the rest'),
+        *HITFINDER_OPTIONS,
+)
+class RadonHits(strax.Plugin):
+    """
+    Find hits using the find_hits algorithm in strax.
+    """
+    __version__ = '0.0.7'
+
+    parallel = 'False'
```

Comment:
Can't this be `True`?

Reply:
Not 100% sure about the consequences but I'll change it!